### DISCUSSION TITLE
Trouble building openssl for x86-android

### DISCUSSION THREAD

**Discussion by nathanstouffer** [microsoft/vcpkg#29506] [Q&A]
Hello everyone! I've got an odd issue building openssl for x86-android. I am using `vcpkg` in [manifest mode](https://vcpkg.readthedocs.io/en/latest/users/manifests/) and my dependences for openssl look like this:

```json
"dependencies": [ { "openssl", "platform": "!emscripten & !windows" } ]
"overrides": [ { "name": "openssl", "version-string": "1.1.1i" } ]
```

I build openssl for the following triplets `x64-windows-static`, `arm-android`, `arm64-android`, `x86-android`, and `wasm32-emscripten`. On a fresh clone of the repo, I tried to install my dependencies with `vcpkg`. It succeeded for the first three (windows and android for arm/arm64) but failed on `x86-android`. Here is the output:

```
Installing 1/7 openssl:x86-android...
Building openss[core]:x86-android...
warning: -- Using community triplet x86-android. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: C:\src\vcpkg\triplets\community\x86-android.cmake
-- Installing port from location: C:\src\vcpkg\buildtrees\versioning_\versions\openssl\30228fbd272c8554937ac94b77fb971249c7c8a
CMake Error at buildtrees/versioning_/versions/openssl/30228fbd272c8554937ac94b77fb971249c7c8a/portfile.cmake:2 (message):
  Can't build opnessl if libressl/boringssl is installed.  Please rmove
  libressl/boringssl, and try install opnessl again if you need it.
Call Stack (most recent call first):
  scripts/ports/cmake:147 (include)
```

I have not explicitly installed `libressl` or `boringssl`, though it could have been installed as an upstream dependency. I looked at the portfile for `30228fb...` and it has the following text that throws the error:

```cmake
if (EXISTS "${CURRENT_INSTALLED_DIIR}/include/openssl/ssl.h")
  message(FATAL_ERROR "Can't build openssl if libressl/boringssl is installed. Please remove libressl/boringssl, and try install openssl again if you need it.")
endif()
```

The file `ssl.h` does exist in that directory, so it seems reasonable that the error would trigger. However, that file is also present in the builds for `arm-android` and `arm64-android`. They also use the same portfile version but build successfully.

Additionally, I have CI set up and ran a completely fresh build (totally ignores all caching), and the build succeeded.

Per [this](https://github.com/microsoft/vcpkg/issues/14164) discussion, have deleted the `vcpkg` archive `AppData`. I have also deleted the archive and my `vcpkg` repo and cloned it entirely from scratch (I don't think there are any other `vcpkg` caches?). But to no avail, still fails on `x86-android` and only on `x86-android`.

Does anyone have advice/thoughts for me?

**Comment by Osyotr** [comment]
I would first look for a port that installed this file:
```
cd installed/vcpkg/info
grep -r --include \*x86-android.list openssl
```

**Reply by nathanstouffer** [reply]
Just ran my install script to inspect this. But now, inexplicably to me, my install succeeds.

The only thing I can think of that might explain this is that I have been in the process of upgrading from VS 2019 to VS 2022. My troubles with vcpkg began when I installed VS2022. And I just finished the upgrade (updating PATH and such)... and now vcpkg for `x86-android` works?

I have no idea how those could be related, but maybe they are. Now I cannot replicate my situation and check in on this.
